client/landing/stepper/declarative-flow/internals/steps-repository/goals/index.tsx

```diff
@@ -98,14 +89,19 @@ const GoalsStep: Step = ( { navigation } ) => {
 	};
 
 	const handleSubmit = ( submittedGoals: Onboard.SiteGoal[] ) => {
-		const intent = handleChange( submittedGoals );
+		setGoals( goals );
```

Comment:
I think maybe we don't need to do this as we always call `setGoals` function whenever the value of the `goals` changes. That being said, the value of the `goals` in the store is always updated to date, and we might not need to do this when submitting. What do you think?

Reply:
I see where you are coming from. Unfortunately there is a hidden complexity to this. Initially all goals were supposed to be `SelectCards`. However due to implications with other screens, `import` and `difm` goals are now displayed as buttons. 
![Screenshot 2022-06-23 at 12 22 17](https://user-images.githubusercontent.com/2019970/175277700-4890ee5d-47c9-48b6-936d-f334c44f692d.png)

Clicking any of those buttons triggers `onSubmit` with an extended goals list, but it does not trigger `onChange`.

Reply:
#### Summarising
> I think maybe we don't need to do this as we always call setGoals function whenever the value of the goals changes

We do not always call `setGoals` function whenever the value of the goals changes.
We call `setGoals` function whenever goals selection changes (checkboxes are set/unset). If `import` or `difm` is selected we receive an updated list of goals as an argument to `onSubmit`.

Reply:
👍